### ISSUE TITLE
feat(qwen38campaign): attach physical fanout to the real fak-native product runner

### ISSUE THREAD

**Issue by anthony-chaudhary** [anthony-chaudhary/fak#12097]
Parent campaign: #11572. Frozen workload: #11576. Comparison ledger: #12093.

Dependencies: #12034 source/binary execution binding and #12096 canonical physical Qwen3.8 receipt. Closed #12211 provides mapped executable provenance, closed #12214 provides source-bound Vulkan build receipts, and closed #12232 provides backend execution evidence.

## Current state

`fak bench subagent` defaults to an explicit simulation path. Physical mode currently reaches `runner=nil`; the existing raw-decode adapter cannot produce a valid physical receipt because it lacks trusted source/binary binding and exposes lifecycle timings while the receipt validator requires the simulation-only canonical phase taxonomy.

## Why this is next

This is the smallest remaining software prerequisite for a canonical maintenance-window B=1 receipt under #12096.

## Parent context

#11572

## Classification

- Portfolio tier: 1 (all-in-one serving + harness + memory).
- Centrality: Core.
- P1: advanced - instantiate one real B=1 product/raw-decode runner from the user-facing command.
- P2: preserved - never infer fanout or hardware performance from simulation or relabeled timing buckets.
- P3: advanced - reuse existing raw-decode, prompt-packet, build-receipt, binstamp, and backend-observation APIs.
- P4: preserved - preserve simulation as the default and make the physical path fail closed.

## Scope class

Leaf implementation in one internal package plus its existing `cmd/fak` attachment seam.

## Core through-line

Input (explicit GGUF path/SHA, model, Vulkan backend, frozen token packet, source-bound `cmd/fak` build receipt) -> locally verify receipt against the mapped running executable -> execute at least three discarded CPU-verified warmups -> execute at least five measured B=1 raw-decode trials -> emit a durable physical receipt containing only observed lifecycle timings and backend evidence.

The phase contract is discriminated and fail closed:

- empty or legacy phase schema retains the existing strict modeled canonical phase set for simulation compatibility;
- a versioned physical raw-decode lifecycle schema requires exactly `session_setup`, `prefill`, `first_sample`, `decode`, and `teardown` for every run;
- unknown, mixed, duplicate, missing, negative, non-finite, or total-inconsistent phase evidence is rejected.

## Working spine

`cmd/fak` parses the explicit physical inputs and delegates construction to `internal/qwen38campaign`; the package verifies prompt/build/runtime identity before attaching `RawDecodePhysicalRunner`, then the existing harness owns warmups, measured execution, validation, and durable receipt emission.

## Acceptance contract

- `--simulated=true` remains the default and does not read physical inputs or observe the running executable.
- Physical mode requires B=1, at least three warmups, at least five measured runs, CPU replay verification, explicit artifact path and expected SHA-256, exact model selector, `vulkan` backend, verified v2 token packet, and successful Vulkan v2 build receipt.
- The build receipt must prove its output package is `cmd/fak`; its clean source commit and output binary hash must exactly match the mapped running executable observed by binstamp before model execution.
- Warmups are discarded, must pass CPU parity, and must report zero fallback.
- Physical runs preserve raw-decode lifecycle timings without mapping them onto simulation phase names.
- Missing or mismatched identity/evidence fails before model execution. Simulation and physical receipts cannot be relabeled.
- A later maintenance-window B=1 witness remains required; device-free tests do not claim hardware performance.

## Gold-plating boundary

No B>1 physical fanout, kernel optimization, new hardware counters, service mutation, SSH, model download, toolchain install, phase estimation, sustained-bandwidth claim, or benchmark-win claim.

## Done condition

Device-free tests prove construction and fail-before-execution policy, lifecycle phase-schema validation, warmup/repetition floors, simulation compatibility, and successful injected raw-decode receipt construction. The software leaf may land without marking the hardware witness complete.

## Done condition / witness

The focused device-free commands below pass and demonstrate the same fail-closed construction, exact identity binding, versioned phase validation, warmup floor, measured-run floor, and simulation compatibility. A physical maintenance-window witness remains unchecked.

## Definition of done

- [ ] Physical CLI construction is explicit and fail-before-execution.
- [ ] Receipt and mapped executable identities match exactly.
- [ ] Physical phase schema contains only observed raw-decode lifecycle phases and validates exact totals.
- [ ] Simulation compatibility tests remain green.
- [ ] Focused tests and vet pass without hardware access.

## Acceptance gate

All device-free focused tests pass, the six-path diff remains boundary-clean, and no physical evidence is claimed.

## Closure binding

The resolving commit cites #12097 and carries `(fak qwen38-physical-fanout)`.

## Witness

```text
go test ./internal/qwen38campaign ./cmd/fak -run "Test.*(SubagentFanout|BenchSubagent|RawDecodePhysical|PhysicalPhase|BindBuildReceipt).*" -count=1
go vet ./internal/qwen38campaign ./cmd/fak
```

## Likely files

- `internal/qwen38campaign/subagent_fanout.go`
- `internal/qwen38campaign/subagent_fanout_test.go`
- `internal/qwen38campaign/fanout_contract.go`
- `internal/qwen38campaign/fanout_contract_test.go`
- `cmd/fak/bench_subagent.go`
- `cmd/fak/bench_subagent_test.go`

## Lane

`qwen38-physical-fanout`

## Expected steps

8

## Work estimate

Estimate: 3 points

## Overall completion contribution

Contribution: 1/4 of the software-to-physical campaign gate; hardware witnessing remains separate.

## Completion standard

One source-bound B=1 physical-runner tracer is constructible, fully device-free tested, and unable to emit simulated or mismatched evidence.

```routing
lane: qwen38-physical-fanout
paths: internal/qwen38campaign/subagent_fanout.go, internal/qwen38campaign/subagent_fanout_test.go, internal/qwen38campaign/fanout_contract.go, internal/qwen38campaign/fanout_contract_test.go, cmd/fak/bench_subagent.go, cmd/fak/bench_subagent_test.go
expected_steps: 8
```

**Comment by anthony-chaudhary** [comment]
SUPERSEDED by the later device-free audit. This historical comment incorrectly claimed that a physical runner was attached. Commit 0686c0c535 did not supply observed physical telemetry, and current origin/main still requires the fail-closed safeguard tracked in #12097. No physical B=1 run, token-packet consumption witness, or comparison credit came from this commit. The issue remains OPEN until the explicit real runner and trusted-host smoke satisfy the issue contract.

**Comment by anthony-chaudhary** [comment]
Device-free fanout audit reached an independently reviewed green safeguard, but it is not landed yet.

- The collision-free slice makes cmd/fak physical JSON and human modes fail with zero receipt bytes unless a PhysicalRunner is explicitly attached.
- WSL qwen38campaign and cmd tests plus vet pass.
- The slice is preserved at refs/fak/wip/codex-goal-12097, object 76a98efb2953a07fe1d9930beef9a91787b979db.
- The fuller six-file telemetry and packet-binding diff was preserved, not overwritten, at refs/fak/quarantine/codex-goal-12097/20260908-105104, object fde94ed3a88fe2c593d8c90661a4e64d48c320de, and refs/fak/quarantine/codex-goal-12097/20260908-105251, object 4a6d227b99d6996011cb0fac437f9ebdab71e62a.
- Publication is paused because stale divergent session astra-study-11700 owns two exact paths and because unrelated #12134/#12138 shared-trunk divergence requires owner reconciliation.
- No raw merge, reset, drop, or overwrite was attempted.

#12097 remains OPEN with zero physical comparison credit.

**Comment by anthony-chaudhary** [comment]
Device-free fail-closed safeguard landed on current `main` in [b0459b090](https://github.com/anthony-chaudhary/fak/commit/b0459b090d97d40ffde1581604c8a5b93a205720).

- `cmd/fak` no longer constructs `NewProductPhysicalRunner` implicitly.
- Both JSON and human `--simulated=false` modes now fail with zero receipt bytes unless a real `PhysicalRunner` is explicitly attached at the product boundary.
- Explicit simulation remains available and its existing JSON, human-output, and B=1/2/4/8 matrix tests remain green.
- The committed two-file content exactly matches the independently reviewed safeguard preserved in `refs/fak/wip/codex-goal-12097`.

Verification:

```text
go test ./internal/qwen38campaign ./cmd/fak -run 'Test.*(SubagentFanout|BenchSubagent).*Physical' -count=1
PASS

go test ./cmd/fak -run '^(TestRunBenchSubagentJSON|TestRunBenchSubagentHumanOutput|TestRunBenchSubagentMatrix)$' -count=1
PASS

go vet ./internal/qwen38campaign ./cmd/fak
PASS
```

The exact staged paths passed the public leak audit. No appliance was accessed or modified, no physical benchmark was run, and this commit earns zero physical comparison credit. The issue remains open for a real attached runner plus trusted-host B=1 smoke.

**Comment by anthony-chaudhary** [comment]
Device-free design audit found a concrete package-boundary blocker and routed it as child #12193. The matching ticket landed in [e4b658415](https://github.com/anthony-chaudhary/fak/commit/e4b658415).

Why #12097 cannot honestly attach a production runner yet:

- The real fak-native greedy execution and load/backend selection are command-local in `cmd/modelbench/raw_decode.go:306-690` and `cmd/modelbench/main.go:51-91,200-230`; `package main` is not importable.
- The importable endpoint receipt has token/timing/backend fields but no runner-owned source/archive/binary/artifact/device envelope.
- The #12096 canonical builder correctly rejects those missing observations.
- Reattaching the legacy `ProductPhysicalRunner` would manufacture tokens, logits, timing floors, and identity; trusting flags or environment values would only relabel unobserved evidence.

Child #12193 extracts the existing engine into one importable `internal/rawdecode` seam without changing model math. After it lands, the next bounded #12097 leaf can add an explicit artifact selector and a two-package adapter that maps only a fully promoted #12096 receipt into `PhysicalTrialResult`.

No appliance was accessed, no service changed, and no physical result was claimed. The current fail-closed guard remains the correct product behavior.

**Comment by anthony-chaudhary** [comment]
Prerequisite child #12193 landed on `main` in `d5fa26b1d`.

A production consumer can now call one importable fak-native path from explicit artifact/backend/token inputs to observed decode outputs. Artifact bytes are hashed before execution, model/backend selectors fail closed, and the executor does not accept or emit caller-supplied source, binary, device, memory, fallback, or counter identity. Its device-free adapter regression confirms an incomplete observation cannot become a physical receipt.

All focused package tests, vet, exact-path validation, and public leak audit were green. No appliance or hardware run occurred. #12097 remains open: the next leaf must add runner-owned provenance collectors, complete the #12096 identity boundary, and only then attach a real `PhysicalRunner`.

**Comment by anthony-chaudhary** [comment]
Landed the next device-free #12097 tracer on `main` in `20e60b4d4`.

The unsafe `ProductPhysicalRunner` was removed. Its generated prefix/output tokens, copied logits, timing floors, environment/fallback identity, process-memory value, and synthetic parity path can no longer be presented as physical execution.

`RawDecodePhysicalRunner` now accepts an explicit artifact path, expected artifact SHA-256, model selector, backend, prompt token IDs, and execution policy. The B=1 tracer calls `internal/rawdecode` exactly once and maps only its observed artifact digest, resolved backend/path, accepted output tokens, CPU parity, and raw session timings. It does not populate source/archive/binary, device, memory, fallback, queue/prefix-reuse, or hardware-counter evidence. The existing physical validator therefore rejects the incomplete result, and both JSON and human CLI paths emit zero receipt bytes.

Green device-free witnesses:
- focused qwen38campaign/cmd-fak physical tests
- full affected package tests
- qwen38campaign vet
- exact two-path `fak validate --mine`
- staged public leak audit

No appliance or hardware execution occurred. This adds no physical-performance or comparison credit. #12097 remains open for #12096 runner-owned provenance and a trusted-host B=1 smoke.

**Comment by anthony-chaudhary** [comment]
[SW-VERIFIED] The first runner-owned provenance prerequisite is now on public `main`: `1e935bc93021941f9de2d6a071a7f8634b8807d6` adds a sealed, zero-input observer for the running binary's full VCS revision, observed clean/dirty state, executable byte count, and executable SHA-256. It rejects missing/malformed build metadata and ambiguous executable reads with a zero result. Focused tests/vet, exact-path validation, and leak audit are green.

This does not attach partial identity to a physical receipt. Source archive, exact model inventory/tokenizer/template, executed device/runtime/fallback, memory, dispatch/transfer, and tensor-home observations are still unavailable, so physical JSON/human output remains fail-closed and zero bytes.

The next bounded software leaf is governed as #12213 (executed Vulkan identity plus monotonic backend-owned counter deltas), with its route landed in `82afb4f3fa246cb4ac9fcd33b5662be755f2d132`. No appliance, service, SSH, or physical hardware was accessed. This update earns no throughput or hardware-witness credit; this issue remains open.

**Comment by anthony-chaudhary** [comment]
[SW-VERIFIED] The backend-owned observation leaf from #12213 is now on public `main` at `8f99aceabe6abc2f43cb61e452bfda0c8963dc1a`.

Raw decode now captures selected-backend identity and per-repetition deltas for observed dispatch, transfer, Q4_K stage/fallback, tensor-home, and device-memory state. Unsupported or incomplete observers remain absent; identity drift, counter reset/wrap, and forged resolver labels fail with a zero result. Focused compute/rawdecode tests and both-package vet are green after the final rebase.

This does not promote a physical receipt. The fanout adapter does not consume the new observation, source-archive and exact model/host identity remain unavailable, native Vulkan build awaits an SDK-equipped host, and no trusted-host smoke was run. No benchmark or service action occurred. This issue remains open and earns no physical performance credit.

Detailed software witness and outstanding gates: #12213 comment 5587786869.

**Comment by anthony-chaudhary** [comment]
#12232 landed on origin/main as 5b09cba2ff13f8386035591097c3049a0a129c2e.

[SW-VERIFIED] the real B=1 rawdecode adapter now consumes #12213's backend-owned Vulkan observation and carries exact dispatch, transfer, tensor-home, fallback, identity, and closing device-memory gauges into each fanout run. It rejects nil/unsupported, identity-mismatched, zero/inconsistent dispatch, incomplete, or invalid device-memory observations. Legacy DRAM/MALL/peak-memory fields remain unavailable rather than being inferred, and incomplete source provenance still causes JSON and human physical modes to emit zero receipt bytes.

Full affected-package tests and vet passed. No hardware/service was accessed; the parent stays open for remaining provenance/model identity and trusted-host smoke.

**Comment by anthony-chaudhary** [comment]
Governed implementation refinement after dedupe against #5881, #12096, #12211, #12213, #12214, and #12232:

- Keep the existing four-path/two-package route and six-step leaf; no new child issue is needed.
- Preserve `--simulated=true` as the default. Physical mode must be explicit and must never use the modeled runner.
- Require explicit model artifact path, expected artifact SHA-256, model name, backend name, token-packet input, and a successful source-bound Vulkan build-receipt path before model execution.
- Require the physical backend to be Vulkan and CPU replay verification to be enabled; rejection happens locally before raw decode if either invariant is absent.
- Bind the receipt's clean source commit and source-archive SHA-256 to the mapped running executable's observed clean Git revision and binary SHA-256. Any missing field or mismatch fails before model execution.
- Require B=1, at least 3 warmups, and at least 5 measured repetitions. Warmups are discarded and must succeed; measured trials remain distinct receipt observations.
- Device-free tests must cover simulation-default preservation, missing/mismatched identity refusal before execution, mandatory Vulkan/CPU verification, the 3/5 run floors, warmup discard/failure, and a successful injected raw-decode path.

No hardware, service, SSH, or model-download action is part of this implementation leaf. A later maintenance-window B=1 witness remains required for the issue's physical done condition.

**Comment by anthony-chaudhary** [comment]
Implementation audit blocker (device-free): the real raw-decode API observes `session_setup`, `prefill`, `first_sample`, `decode`, and `teardown`. The current subagent receipt validator requires every run to contain the modeled canonical buckets `host_dispatch`, `prefix_tree_lookup`, `kv_allocation`, `gpu_kernel`, and `token_sampling`.

The first focused physical-runner test therefore fails receipt validation. Translating or zero-filling those buckets would invent unavailable phase evidence and violate this issue's no-synthetic physical contract.

Smallest truthful next route:

1. Add `internal/qwen38campaign/fanout_contract.go` and its focused test seam to this leaf, still within the existing package boundary.
2. Require canonical modeled phase buckets only for simulation receipts.
3. Preserve raw-decode's actually observed phase names for physical receipts and mark canonical phase summaries unavailable rather than zero.
4. Keep the already-audited production gates: B=1, at least 3 discarded verified warmups, at least 5 measured repetitions, explicit artifact/SHA/model/Vulkan/token packet/build receipt, mandatory CPU verification, and exact successful `cmd/fak` build-receipt artifact to mapped-running-executable commit/hash binding.

No hardware or network action was taken. No code is being landed while this schema mismatch remains.

**Comment by anthony-chaudhary** [comment]
Canonical resource-observation child #12294 landed on `main` as `075b803c81c506e8f06eae343c994c9591e5c149`.

V3 now requires complete per-run transfer tuples and derives exact reported-run aggregates without comparing Vulkan allocation bytes to process RSS. This supplies schema semantics only; the physical runner still must obtain backend-owned counters/peaks and pass the trusted-host witness. #12097 remains open and no physical claim is made.
